playwright-e2e/tests/dsm/kitUploadFlow/OS_PE-CGS-saliva-kit-upload-send-receive.spec.ts

```diff
@@ -0,0 +1,179 @@
+import {test} from '@playwright/test';
+import {WelcomePage} from 'pages/dsm/welcome-page';
+import HomePage from 'pages/dsm/home-page';
+import {Navigation} from 'lib/component/dsm/navigation/navigation';
+import {login} from 'authentication/auth-dsm';
+import {StudyEnum} from 'lib/component/dsm/navigation/enums/selectStudyNav-enum';
+import ParticipantListPage from 'pages/dsm/participantList-page';
+import {StudyNavEnum} from 'lib/component/dsm/navigation/enums/studyNav-enum';
+import ParticipantPage from 'pages/dsm/participant-page/participant-page';
+import {KitUploadInfo} from 'pages/dsm/kitUpload-page/models/kitUpload-model';
+import ContactInformationTab from 'lib/component/dsm/tabs/contactInformationTab';
+import {TabEnum} from 'lib/component/dsm/tabs/enums/tab-enum';
+import {SamplesNavEnum} from 'lib/component/dsm/navigation/enums/samplesNav-enum';
+import {KitTypeEnum} from 'lib/component/dsm/kitType/enums/kitType-enum';
+import KitUploadPage from 'pages/dsm/kitUpload-page/kitUpload-page';
+import InitialScanPage from 'pages/dsm/initialScan-page';
+import FinalScanPage from 'pages/dsm/finalScan-page';
+import crypto from 'crypto';
+import SampleInformationTab from 'lib/component/dsm/tabs/sampleInformationTab';
+import {SampleInfoEnum} from 'lib/component/dsm/tabs/enums/sampleInfo-enum';
+import {SampleStatusEnum} from 'lib/component/dsm/tabs/enums/sampleStatus-enum';
+import KitsWithoutLabelPage from 'pages/dsm/kitsInfo-pages/kitsWithoutLabel-page';
+import {KitsColumnsEnum} from 'pages/dsm/kitsInfo-pages/enums/kitsColumns-enum';
+import KitsSentPage from 'pages/dsm/kitsInfo-pages/kitsSentPage';
+import KitsReceivedPage from 'pages/dsm/kitsInfo-pages/kitsReceived-page/kitsReceivedPage';
+
+
+test.describe('Saliva Kits upload flow (OC PE-CGS)', () => {
+  let welcomePage: WelcomePage;
+  let homePage: HomePage;
+  let navigation: Navigation;
+
+  let shortID: string;
+  let kitUploadInfo: KitUploadInfo;
+  let kitLabel: string;
+  let shippingID: string;
+
+  const kitType = KitTypeEnum.SALIVA;
+  const expectedKitTypes = [KitTypeEnum.SALIVA, KitTypeEnum.BLOOD];
+
+  test.beforeEach(async ({ page, request }) => {
```

Comment:
I don't know why you created two `test.beforeEach`. Here and line 48.

Reply:
The first assigns and does initial preparation and the second calls methods on the initial (home) page.